### ISSUE TITLE
Cannot Clone bitmap using PixelFormat.Format4bppIndexed on linux

### ISSUE THREAD

**Issue by zappyzhao** [dotnet/corefx#37209]
Simple code:
`var originMap = new Bitmap(1, 1);`
`var cloneMap = originMap.Clone(new Rectangle(0, 0, 1, 1), PixelFormat.Format4bppIndexed);`
`Console.WriteLine("PixelFormat of cloneMap: " + cloneMap.PixelFormat.ToString());`

If I run this on windows (x64, windows 10 Enterprise), the result shows as:
`PixelFormat of cloneMap: Format4bppIndexed`

However, I ran the same code on linux environment on the same machine (I ran linux using Docker Desktop for windows), the result:
`PixelFormat of cloneMap: Format32bppArgb`

Could anyone please help on this? Or if there's any workaround? Thanks!

**Comment by gfoidl** [comment]
On platforms other than Windows [libgdiplus](https://github.com/mono/libgdiplus) is used.
If the "bug" can be reproduced on Mono, please move the issue the _libgdiplus_-repo.

**Comment by gfoidl** [comment]
Can repro this on Mono, too -> https://github.com/mono/libgdiplus/issues/541

**Comment by gfoidl** [comment]
Hm, when trying to reproduce this with C and libgdiplus the problem doesn't occur.
So the cause for this must be somewhere else. I'm having a look...

**Comment by gfoidl** [comment]
> when trying to reproduce this with C and libgdiplus the problem doesn't occur

I had a [mistake](https://github.com/mono/libgdiplus/issues/541#issuecomment-487164175) in the test-code. The bug is not in the managed side. There it's just a marshalling to unmanaged code: https://github.com/dotnet/corefx/blob/dfe7593397bf5c598548486c6cb774ebf60e9d69/src/System.Drawing.Common/src/System/Drawing/Bitmap.cs#L176-L183 to the native method `GdipCloneBitmapArea`.

**Comment by gfoidl** [comment]
The fix is here -> https://github.com/mono/libgdiplus/pull/543

**Comment by zappyzhao** [comment]
@gfoidl Thank you so much for the fast investigate and quick fix! May I know how long it will take the code to be checked in and how soon I'll be able to refer to the lastest System.Drawing.Common library please? Since my project is blocked by this now.

**Comment by gfoidl** [comment]
> how long it will take the code to be checked in

I can't answer this, as I'm not part of the team.

> lastest System.Drawing.Common library

The bug is not in `System.Drawing.Common`, so there's no need to update this library.
The bug is in `libgdiplus.so`.

So to unblock your project you could build `libgdiplus.so` by your own and use that version.
Take care that you checkout the branch with the fix 

```bash
git clone https://github.com/gfoidl/libgdiplus.git
cd libgdiplus
git branch bitmap_clone
./autogen.sh
make
make check
make install
```
(hope I typed it right).

**Comment by zappyzhao** [comment]
@gfoidl Thanks a lot! will try this out

**Comment by safern** [comment]
Thanks @gfoidl for helping our here. This indeed looks like a libgdiplus issue and the fix is linked above. Closing as there is no action on our side.
